File: libs/service_layer/search/SearchEngine/src/main/kotlin/net/dankito/service/search/specific/ReferenceSearch.java
package net.dankito.service.search.specific;


import net.dankito.deepthought.model.Reference;
import net.dankito.service.search.SearchBase;
import net.dankito.service.search.SearchCompletedListener;

import java.util.ArrayList;
import java.util.Collection;


public class ReferenceSearch extends SearchBase {

  protected Collection<Reference> results = new ArrayList<>();

  protected SearchCompletedListener<Collection<Reference>> completedListener = null;


  public ReferenceSearch(String searchTerm, SearchCompletedListener<Collection<Reference>> completedListener) {
    super(searchTerm);
    this.completedListener = completedListener;
  }


  public void setResults(Collection<Reference> results) {
    this.results = results;
  }

  protected void callCompletedListener() {
    if(completedListener != null)
      completedListener.completed(results);
  }

  @Override
  protected int getResultsCount() {
    return results.size();
  }


  public boolean addResult(Reference result) {
    return this.results.add(result);
  }

  public Collection<Reference> getResults() {
    return results;
  }


}

File: libs/service_layer/search/SearchEngine/src/main/kotlin/net/dankito/service/search/specific/ReferenceSearch.kt
package net.dankito.service.search.specific


import net.dankito.deepthought.model.FileLink
import net.dankito.deepthought.model.Series
import net.dankito.deepthought.model.Source
import net.dankito.service.search.Search
import net.dankito.service.search.SearchWithCollectionResult


class ReferenceSearch(searchTerm: String = Search.EmptySearchTerm,
                      val mustHaveThisSeries: Series? = null, val mustHaveTheseFiles: Collection<FileLink> = mutableListOf(),
                      completedListener: (List<Source>) -> Unit)
    : SearchWithCollectionResult<Source>(searchTerm, completedListener) {

}
